File: hex.js


function tohex(num)
{
    var hex = [];
    while(num > 0)
    {
        if (num % 16 > 9)
        {
            hex.push(String.fromCharCode(55 + num % 16));
        }
        else
        {
            hex.push(num % 16);
        }
        num = Math.floor(num / 16);
    }
    return hex.reverse().join("");
}
console.log(tohex(20));

File: hex.dart
String toHex(num n) {
  List<String> hex = [];
  while (n > 0) {
    int remainder = n.toInt() % 16;
    if (remainder > 9) {
      hex.add(String.fromCharCode(55 + remainder));
    } else {
      hex.add(remainder.toString());
    }
    n = (n / 16).floor();
  }
  hex = hex.reversed.toList();
  var out = hex.join();
  
  return out.isEmpty ? '0' : out;
}

void main() {
  print(toHex(5));   // Expected output: 5
  print(toHex(10));  // Expected output: A
  print(toHex(55));  // Expected output: 37
}
